predelete/predelete.go

```diff
@@ -62,6 +67,33 @@ func PreDeleteOperatorSecret(clientset *kubernetes.Clientset) error {
 	return nil
 }
 
+func PreDeleteEnterprisePortalSecret(clientset *kubernetes.Clientset) error {
+	fmt.Println("Running pre delete hook")
+	secrets, err := clientset.CoreV1().Secrets(os.Getenv("TYK_POD_NAMESPACE")).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	found := false
+	for _, value := range secrets.Items {
+		if value.Name == os.Getenv("ENTERPRISE_PORTAL_SECRET_NAME") {
+			err = clientset.CoreV1().Secrets(os.Getenv("TYK_POD_NAMESPACE")).Delete(context.TODO(), value.Name, metav1.DeleteOptions{})
+			if err != nil {
+				return err
+			}
+			found = true
+			break
+		}
+	}
+
+	if found == false {
+		fmt.Println("A previously created enterprise portal secret has not been identified")
+	} else {
+		fmt.Println("A previously created enterprise portal secret was identified and deleted")
+	}
+	return nil
+}
```

Comment:
Not crucial but I would prefer early return statements. If it looks okay to you, we can update it as follows:

```suggestion
	if !found {
		fmt.Println("A previously created enterprise portal secret has not been identified")

		return nil
	}

	fmt.Println("A previously created enterprise portal secret was identified and deleted")
	return nil
```